src/main/java/com/linecorp/armeria/client/routing/NodeSelectStrategy.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.routing;
+
+/**
+ * A factory interface of {@link NodeSelector}
+ */
+public interface NodeSelectStrategy<T extends Node> {
+    enum SelectStrategyType {
```

Comment:
is it like this?
```
public interface NodeSelectStrategy<T extends Node> {
    RoundRobinStrategy ROUND_ROBIN_STRATEGY = new RoundRobinStrategy();

    static RoundRobinStrategy getRoundRobinStrategy(){
        return ROUND_ROBIN_STRATEGY;
    }

    NodeSelector<T> newSelector(ServerGroup<? extends T> serverGroup);

}
```
if yes,when register server group user need invoke this method to choose which NodeSelectStrategy to use.enum literal's mean is straightforward,no need concern about implementation detail.I don't know whether i understand your mean correctly.